gulpfile.js

```diff
@@ -6,32 +6,100 @@ const del = require('del');
 const through = require('through2');
 const gulp = require('gulp');
 const file = require('gulp-file');
-const fontello = require('gulp-fontello');
-const download = require('gulp-download');
-const merge = require('gulp-merge');
 const gutil = require('gulp-util');
 const zip = require('gulp-zip');
 const jsonSchema = require('gulp-json-schema');
 const sourcemaps = require('gulp-sourcemaps');
 const tsc = require('gulp-typescript');
 const tslint = require('gulp-tslint');
 const refresh = require('gulp-refresh');
-const spawn = require('child_process').spawn;
 const appSchema = require('./app-schema.json');
+const argv = require('yargs').argv;
 
 function getFolders(dir) {
     return fs.readdirSync(dir).filter((file) => fs.statSync(path.join(dir, file)).isDirectory());
 }
 
+function slugify(text) {
+  return text.toString().toLowerCase()
+    .replace(/^\s+|\s+$/g, '')
+    .replace(/\s+/g, '-')           // Replace spaces with -
+    .replace(/[^\w-]+/g, '')       // Remove all non-word chars
+    .replace(/--+/g, '-')         // Replace multiple - with single -
+    .replace(/^-+/, '')             // Trim - from start of text
+    .replace(/-+$/, '');            // Trim - from end of textte
+}
+function sanitize(text) {
+    const safeString = text.toString()
+    .replace(/^\s+|\s+$/g, '')
+    .replace(/\s+/g, '')
+    .replace(/[^\w-]+/g, '')       // Remove all non-word chars
+    .replace(/-+/g, '')         // Replace multiple - with single -
+    .replace(/^-+/, '')             // Trim - from start of text
+    .replace(/-+$/, '');            // Trim - from end of textte
+
+    return `${ safeString[0].toUpperCase() }${ safeString.substr(1).toLowerCase() }`;
+}
+
 const appsPath = './apps';
 const tsp = tsc.createProject('tsconfig.json');
 
 gulp.task('clean-generated', function _cleanTypescript() {
-    return del(['./dist/**', './.server-dist/**']);
+    return del(['./dist/**']);
 });
 
-gulp.task('clean-fontello', function _cleanFontello() {
-    return del(['./.site/fontello/**']);
+gulp.task('create-app', function _createNewApp() {
+    const slugifiedName = slugify(argv.name);
+    if(typeof argv.name != 'string' || argv.name === undefined) {
+      gutil.log(gutil.colors.red(figures.cross),  `Please use ${ gutil.colors.cyan('npm run create-app <name>') } to create a new Rocket.Chat App`);
+      throw new Error('Incorrect usage of create-app command');
+    }
+    if(fs.existsSync(`./apps/${ slugifiedName }`)) {
+      gutil.log(gutil.colors.red(figures.cross),  `${ gutil.colors.cyan(argv.name) } already exists in the apps folder`);
```

Comment:
The folder already exists, so shouldn't we return out?

Reply:
Honestly, I personally would rather us have a `return` that way we exit as soon as possible and thus the flow is easier to understand at a glance.